service/asset_allocations_service.go

```diff
@@ -24,7 +25,27 @@ func NewAssetAllocationService(repo repository.AssetAllocationsRepository) Asset
 }
 
 func (service *assetAllocationsService) CreateAssetAllocation(ctx context.Context, req contract.CreateAssetAllocationRequest) (*domain.AssetAllocations, error) {
-	return nil, nil
+	assetAllocation, err := service.assetAllocationsRepo.CreateAssetAllocation(ctx, req)
+
+	if err != nil {
+		if err == customerrors.UserNotExist {
```

Comment:
no need of so many conditions , just return if err != nil { return nil, errr} , you are already sending custom error from repository layer